File: spankbang.js
const axios = require('axios');
const cheerio = require('cheerio');

module.exports = async function(query) {
  const url = `https://spankbang.party/s/${encodeURIComponent(query)}`;
  const results = [];

  try {
    const { data } = await axios.get(url, {
      headers: {
        'User-Agent': 'Mozilla/5.0'
      }
    });

    const $ = cheerio.load(data);

    $('div.video-list > a').each((_, el) => {
      const href = $(el).attr('href');
      const title = $(el).find('.title').text().trim();
      const preview = $(el).find('img').attr('data-src') || $(el).find('img').attr('src');
      const duration = $(el).find('.dur').text().trim();

      if (href && title && preview) {
        results.push({
          title,
          url: `https://spankbang.party${href}`,
          preview: preview.startsWith('http') ? preview : `https:${preview}`,
          duration,
          source: 'SpankBang'
        });
      }
    });

    return results;
  } catch (err) {
    console.error('❌ SpankBang scraper error:', err.message);
    return [];
  }
};

File: spankbang.py
# spankbang.py
import aiohttp
from bs4 import BeautifulSoup

async def scrape_spankbang(query):
    search_url = f"https://spankbang.com/s/{query.replace(' ', '+')}/1/"
    headers = {
        "User-Agent": "Mozilla/5.0"
    }
    async with aiohttp.ClientSession() as session:
        async with session.get(search_url, headers=headers) as response:
            if response.status != 200:
                return []
            html = await response.text()
            soup = BeautifulSoup(html, "html.parser")
            results = []
            for video in soup.select(".video-item"):
                title_tag = video.select_one(".title")
                if not title_tag:
                    continue
                title = title_tag.get_text(strip=True)
                href = video.get("href")
                preview = video.select_one("img")["src"] if video.select_one("img") else ""
                results.append({
                    "title": title,
                    "url": f"https://spankbang.com{href}",
                    "preview": preview
                })
            return results
